File: java-order/src/main/java/com/veben/javatokotlin/java/order/domain/PassedOrder.java
package com.veben.javatokotlin.java.order.domain;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import java.time.LocalDateTime;
import java.util.UUID;

import static lombok.AccessLevel.NONE;

@Getter
@Setter
@EqualsAndHashCode(of = "id")
@ToString
@Entity
@Table(name = "passed_order")
public class PassedOrder {

    @Id
    @Setter(NONE)
    private UUID id;

    @Embedded
    private Buyer buyer;

    LocalDateTime date;

    protected PassedOrder() {
        this.id = UUID.randomUUID();
    }

    public PassedOrder(Buyer buyer, LocalDateTime date) {
        this();
        this.buyer = buyer;
        this.date = date;
    }
}

File: kotlin-order/src/main/kotlin/com/veben/microservices/order/domain/PassedOrder.kt
package com.veben.microservices.order.domain

import java.time.LocalDateTime
import java.util.*
import javax.persistence.Embedded
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "passed_order")
data class PassedOrder(
    @Id
    val id: UUID = UUID.randomUUID(),
    @Embedded
    var buyer: Buyer? = null,
    var date: LocalDateTime? = null
)
